ProcessLib/BoundaryCondition/UniformDirichletBoundaryCondition.cpp

```diff
@@ -11,70 +11,57 @@
 
 #include <algorithm>
 #include <vector>
-
 #include <logog/include/logog.hpp>
 
+#include "MeshGeoToolsLib/MeshNodeSearcher.h"
+#include "BoundaryConditionConfig.h"
+
 namespace ProcessLib
 {
-UniformDirichletBoundaryCondition::UniformDirichletBoundaryCondition(
-    GeoLib::GeoObject const& geometry, BaseLib::ConfigTree const& config)
-    : _geometry(geometry)
+std::unique_ptr<UniformDirichletBoundaryCondition>
+createUniformDirichletBoundaryCondition(
+    BoundaryConditionConfig const& config,
+    MeshGeoToolsLib::MeshNodeSearcher& searcher,
+    NumLib::LocalToGlobalIndexMap const& dof_table,
+    int const variable_id)
 {
     DBUG("Constructing UniformDirichletBoundaryCondition from config.");
     //! \ogs_file_param{boundary_condition__type}
-    config.checkConfigParameter("type", "UniformDirichlet");
+    config.config.checkConfigParameter("type", "UniformDirichlet");
```

Comment:
Just an idea: Would deriving BCConfig from ConfigTree be possible, and would it be nice too?